staging/src/k8s.io/apimachinery/pkg/types/namespacedname.go

```diff
@@ -43,3 +43,16 @@ const (
 func (n NamespacedName) String() string {
 	return fmt.Sprintf("%s%c%s%c%s", n.Tenant, Separator, n.Namespace, Separator, n.Name)
 }
+
+// NamespacednameWithTenantSource is used to identify a service and endpoint in scale out architecture
```

Comment:
is this also representing a service in scale-up arch as well?

Reply:
In scale-up arch, there is single source of tenant. It does not hurt to have an extra const field. For better understanding in scale out arch, I prefer to specifically call it out. It should not cause confusing in scale up.